.github/workflows/update-lock-files.yml

```diff
@@ -0,0 +1,69 @@
+# Workflow to update lock files
+name: Update lock files
+
+on:
+  workflow_dispatch:
+
+jobs:
+  update_lock_files:
+    if: "github.repository == 'scikit-learn/scikit-learn' || github.event_name == 'workflow_dispatch'"
+    runs-on: ubuntu-latest
+
+    strategy:
+      # Ensure that each build will continue even if one build in the matrix fails
+      fail-fast: false
+      matrix:
+        include:
+          - name: main
+            update_script_args: "--skip-build 'scipy-dev|^py39_conda_forge$|pypy'"
+            additional_commit_message: "[doc build]"
+          - name: scipy-dev
+            update_script_args: "--select-build scipy_dev"
+            additional_commit_message: "[scipy-dev]"
+          - name: cirrus-arm
+            update_script_args: "--select-build '^py39_conda_forge$'"
+            additional_commit_message: "[cirrus arm]"
+          - name: pypy
+            update_script_args: "--select-build pypy"
+            additional_commit_message: "[pypy]"
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Generate lock files
+        run: |
+          source build_tools/shared.sh
+          source $CONDA/bin/activate
+          conda install -n base conda conda-libmamba-solver -y
+          conda config --set solver libmamba
+          conda install -c conda-forge "$(get_dep conda-lock min)" -y
+
+          python build_tools/update_environments_and_lock_files.py ${{ matrix.update_script_args }}
+      - name: Create Pull Request
+        id: cpr
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ secrets.BOT_GITHUB_TOKEN }}
+          commit-message: Update CI lock files ${{ matrix.additional_commit_message }}
+          committer: "Lock file bot <noreply@github.com>"
+          author: "Lock file bot <noreply@github.com>"
+          delete-branch: true
+          branch: auto-update-lock-files-${{ matrix.name }}
+          title: ":lock: :robot: CI Update lock files for ${{ matrix.name }} CI build(s) :lock: :robot:"
```

Comment:
What happens at that step if the diff is empty? For instance where the lock files are already up to date?

- does it create an empty commit PR?
- does the workflow fail?

Ideally, I would like neither option but a message that says that there are no changes to commit and therefore no pull request has been created.

Reply:
No PR is created if there are no changes and the workflow does not fail.

The `peter-evans/create-pull-request` hash some log that you can look at with all the details:
- [build log](https://github.com/lesteve/scikit-learn/actions/runs/6623755509/job/17991374979) existing PR which needs to be updated you see at one point
```
Pushing pull request branch to 'origin/auto-update-lock-files-scipy-dev'
```
-  [build log](https://github.com/lesteve/scikit-learn/actions/runs/6624326012/job/17993070486) no change needed, no PR created you don't see the `Pushing pull request branch` step